build.js

```diff
@@ -35,16 +35,58 @@ async function rmDistDir() {
   }
 }
 
-async function main() {
-  console.log('Compiling sequelize...');
-  const [declarationFiles, filesToCompile] = await Promise.all([
+function sourcePathToDestPath(source) {
+  return path.resolve(outdir, source).replace(/\.ts$/, '.js');
+}
+
+async function ageCheckFiles(paths) {
+  return (await Promise.all(
+    paths.map(source => new Promise(resolve => {
+      const dest = sourcePathToDestPath(source);
+      const sourceStat = fs.statSync(source, { throwIfNoEntry: false });
+      const destStat = fs.statSync(dest, { throwIfNoEntry: false });
+      const ok = sourceStat && (!destStat || destStat.mtimeMs < sourceStat.mtimeMs);
+      resolve(ok ? source : null);
+    })))).filter(Boolean);
+}
+
+async function listFilesToGenerate() {
+  let toCompile, rmPromise;
+
+  const toCompilePromise = glob('./lib/**/*.[tj]s', { onlyFiles: true, absolute: false });
+
+  if (process.argv.indexOf('--check-mtime') >= 0) {
```

Comment:
Can you add an example on the top of the file on how to use it?

Reply:
Is it good enough with the entry in package.json ?
Else, it would be a bit awkward to write a single comment, when there should actually be a header there a bit more verbose that explains what the script does.

Reply:
done

Reply:
obsolete